File: bases/from_base.py
from consts import numbers_map

inverted_map = {v: k for k, v in numbers_map.items()}


def from_base(number: str, base: int = 2) -> int:
    result = 0
    number = number.replace(" ", "")
    number = reversed([i for i in number.upper()])
    for i, v in enumerate(number):
        char = int(inverted_map[v])
        result += char * (base**i)
    return result


# print(from_base("309b 66f6 88bf ad72", 16)) # /=> 3502506344252943730

File: typescript/bases/from_base.ts
import NUM_MAP from "./consts";

const INVERTED_MAP: Map<string, number> = new Map();

for (let data of NUM_MAP) {
  INVERTED_MAP.set(data[1], data[0]);
}

function from_base(number_: string, base: number = 2): number {
  let result = 0;
  let number_list = number_
    .split("")
    .map((x) => x.toUpperCase())
    .reverse();

  for (let i = 0; i < number_list.length; i++) {
    let v = number_list[i];
    let char = INVERTED_MAP.get(v);

    //@ts-ignore
    result += char * Math.pow(base, i);
  }

  return result;
}